src/vaadin-progress-mixin.ts

```diff
@@ -0,0 +1,92 @@
+import { Constructor } from '@vaadin/mixin-utils';
+import { LitElement, property, PropertyValues } from 'lit-element';
+
+export interface ProgressMixinInterface {
+  value: number | null | undefined;
+  min: number;
+  max: number;
+  indeterminate: boolean;
+}
+
+export const ProgressMixin = <T extends Constructor<LitElement>>(base: T): T & Constructor<ProgressMixinInterface> => {
+  class Progress extends base {
+    @property({ type: Number })
+    value: number | null | undefined;
+
+    @property({ type: Number })
+    min = 0;
+
+    @property({ type: Number })
+    max = 1;
+
+    @property({ type: Boolean, reflect: true })
+    indeterminate = false;
+
+    protected firstUpdated() {
+      this.setAttribute('role', 'progressbar');
+    }
+
+    protected updated(props: PropertyValues) {
+      super.updated(props);
+
+      const minChanged = props.has('min');
+      if (minChanged) {
+        this.min && this._minChanged(this.min);
+      }
+
+      const maxChanged = props.has('max');
+      if (maxChanged) {
+        this.max && this._maxChanged(this.max);
+      }
+
+      const valueChanged = props.has('value');
+      if (valueChanged) {
+        this.value && this._valueChanged(this.value);
+      }
+
+      if (valueChanged || minChanged || maxChanged) {
+        this.value && this.min && this.max && this._normalizedValueChanged(this.value, this.min, this.max);
```

Comment:
Please change this to:

```suggestion
        this._normalizedValueChanged(this.value, this.min, this.max);
```

Reply:
same with this: `this.value !== null && this.value !== undefined &&....`